modules/phase_field/include/kernels/MatGradSqCoupled.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MATGRADSQCOUPLED_H
+#define MATGRADSQCOUPLED_H
+
+#include "Kernel.h"
+#include "JvarMapInterface.h"
+#include "DerivativeMaterialInterface.h"
+
+// Forward Declarations
+class MatGradSqCoupled;
+
+template <>
+InputParameters validParams<MatGradSqCoupled>();
+
+/*
+ * This kernel calculates the prefactor * nabla_psi term in A-C equation for phase field modeling of oxidation
+ * prefactor * grad_psi * grad_psi, where psi is the electrical field variable.
+ * prefactor = 0.5 * grad_permitivity(phi), described in [Materials] using [DerivativeParsedMaterials]
```

Comment:
it is not `grad_permittivity` since it is not a spatial derivative, instead it should be `d_permittivity/d_phi`